packages/sdk/src/types/VersionedModuleSource.ts

```diff
@@ -109,33 +108,61 @@ export async function parseModuleInterface(
  * Extract the embedded smart contract schema bytes. Returns `null` if no schema is embedded.
  * @param {VersionedModuleSource} moduleSource The smart contract module source.
  * @returns {RawModuleSchema | null} The raw module schema if found.
+ * @throws If the module source cannot be parsed or contains duplicate schema sections.
  */
-export async function getEmbeddedModuleSchema(
-    moduleSource: VersionedModuleSource
-): Promise<RawModuleSchema | null> {
-    const wasmModule = await WebAssembly.compile(moduleSource.source);
-    const versionedSchema = schemaBytesFromWasmModule(
-        wasmModule,
-        'concordium-schema'
+export function getEmbeddedModuleSchema({
+    source,
+    version,
+}: VersionedModuleSource): RawModuleSchema | null {
```

Comment:
While we're at it we could also change it to return `undefined` instead of `null`? It seems to be an outlier in returning `null`.

Reply:
Sure, better to be consistent.

Reply:
done